### ISSUE TITLE
Solution to power dataverses, datasets structure visualization on Harvard Dataverse home page

### ISSUE THREAD

**Issue by TaniaSchlatter** [IQSS/dataverse#6750]
There has been a lot of progress made to the visualization for Harvard Dataverse #5603!

We need to enable the visualization to call data from https://dataverse.harvard.edu/, to populate. 
Timing of calls needs to be determined, for example 24 hours, 1 week, 1 month (past 30 days), 1 year.

**Comment by djbrooke** [comment]
Great to hear, @TaniaSchlatter. I'll move this to Ready and we can get it into a sprint soon.

**Comment by scolapasta** [comment]
Discussed at tech hours: as this is just for Harvard Dataverse for now, we decided to keep this simple*. Rather than have an API call (that would need to be triggered by a cron job or timer anyway), we'll just start with this living outside of the main app. There will be python (or other) scripts that will need to live on the server, get called and replace the json file. (and these python scripts will have access to the db). We'll then setup a cron job to call this on a regular basis. And add documentation on how to set up.

All scripts, queries, documentation, etc. should all live in the visualization github project. (we could / should probably move this issue to there for now)

(*) if we find others like this particular visualization, we may consider adding some logic to the app itself, but as others may want variations we'd need to determine a flexible way to do it, so changes could be made without having to fork the project. By doing it the above way, we unblock getting this working in as simple a way as possible, without burdening the app with additional (and possibly unnecessary for many installations) logic.

**Comment by mheppler** [comment]
Data needed in the JSON which is missing from static version...
- parents, childrens, titles, dates (not all have dates currently)
- doi's, file id's for link (missing)
- subjects (nice to have)

**Comment by sekmiller** [comment]
I added a python script which connects to the database, runs a query, transforms the results into json and writes out a file. There is a separate params file which holds the db credentials and denotes whether to use the query params set up for a production or dev db(there are different ids for the dataset field pertaining to dataset subjects.) The json now includes dataset identifiers and dataverse aliases which can be used to create links. Also titles, dates and subjects have been added. The missing piece at this point is setting up a chron job and some documentation.

Updates are being made here: 
https://github.com/IQSS/dataverse-homepage-viz/tree/update-scripts

**Comment by sekmiller** [comment]
Fixed the issue I described yesterday. The "diff" element in the json for a dataverse now contains the difference in days from today back to the most recent publishing date in the dataverse.

**Comment by djbrooke** [comment]
Closing this, as Update scripts IQSS/dataverse-homepage-viz#8 was merged.

**Comment by TaniaSchlatter** [comment]
I see that it was merged, but have we tested the solution to see if it works? I would like to see it and confirm before we close it, and really wanted to see it working before it merged, so I would know the status.

Thanks!

**Comment by djbrooke** [comment]
@TaniaSchlatter Yes, we tested that the data was generated. There were not front end changes expected or implemented as part of this. #5603 is open for tracking the larger work and this was cut out specifically to handle the generation of the backend data on a regular schedule, which is complete. As we discussed last week, we'll come back to the the larger issue of visualization once we have the resources available to take it on. We can track that specific work in the larger open issue or we can create another issue.
